lib/rust/ensogl/component/slider/src/lib.rs

```diff
@@ -0,0 +1,416 @@
+                |(width, height)| Vector2::new(*width, *height)
+            );
+            eval component_size (
+                (size) model.set_size(*size);
+            );
+
+            value_is_default        <- all2(&value, &input.set_value_default).map(
+                |(value, default)| value==default
+            );
+            value_is_default_true   <- value_is_default.on_true();
+            value_is_default_false  <- value_is_default.on_false();
+            eval_ value_is_default_true ({
+                model.value_left.set_property_default(formatting::Weight::Normal);
+                model.value_dot.set_property_default(formatting::Weight::Normal);
+                model.value_right.set_property_default(formatting::Weight::Normal);
+            });
+            eval_ value_is_default_false ({
+                model.value_left.set_property_default(formatting::Weight::Bold);
+                model.value_dot.set_property_default(formatting::Weight::Bold);
+                model.value_right.set_property_default(formatting::Weight::Bold);
+            });
+
+
+            // colors
+
+            background_color.target <+ all2(&input.set_background_color, &input.set_slider_enabled).map(
+                |(color, enabled)| if *enabled { *color } else { color.to_grayscale() }
+            );
+            track_color.target <+ all2(&input.set_slider_color, &input.set_slider_enabled).map(
+                |(color, enabled)| if *enabled { *color } else { color.to_grayscale() }
+            );
+            value_color.target <+ all2(&input.set_value_color, &input.set_slider_enabled).map(
+                |(color, enabled)| if *enabled { *color } else { color.to_grayscale() }
+            );
+            label_color.target <+ all2(&input.set_label_color, &input.set_slider_enabled).map(
+                |(color, enabled)| if *enabled { *color } else { color.to_grayscale() }
+            );
+
+            eval background_color.value (
+                (color) model.set_background_color(*color);
+            );
+            eval track_color.value (
+                (color) model.set_track_color(*color);
+            );
+            eval value_color.value (
+                (color) {
+                    model.value_left.set_property_default(color);
+                    model.value_dot.set_property_default(color);
+                    model.value_right.set_property_default(color);
+                }
+            );
+            eval label_color.value (
+                (color) model.label.set_property_default(color);
+            );
+
+
+            // text alignment
+
+            value_text <- all2(&value, &precision_adjusted).map(
+                |(value, precision)| {
+                    value_text_truncate_precision(*value, *precision)
+                }
+            );
+            value_text_decimal_visible <- value_text.map(|t| t.1.is_some());
+            model.value_left.set_content <+ value_text.map(|t| t.0.clone());
```

Comment:
t.1 and t.0 are hard to understand. If you really need to keep something in a tuple, extract it to well named var first, like `foo <- bar._0;`.